### ISSUE TITLE
Cannot connect to TLSV1.2 endpoint with Envoy proxy

### ISSUE THREAD

**Issue by ryan-motive** [envoyproxy/envoy#20387]
*Description*:
We are building a service that runs with Istio. We are seeing an issue where this service cannot complete the TLSv1.2 handshake with another service. This is what the handshake looks like:

No.	Time	Source	Destination	Protocol	Length	Info
625	8.220923	192.168.70.10	xxx.xx.xxx.xx	TCP	66	59450 → 443 [SYN] Seq=0 Win=64240 Len=0 MSS=1460 WS=256 SACK_PERM=1
660	8.432916	xxx.xx.xxx.xx	192.168.70.10	TCP	66	443 → 59450 [SYN, ACK] Seq=0 Ack=1 Win=64240 Len=0 MSS=1412 SACK_PERM=1 WS=128
661	8.433073	192.168.70.10	xxx.xx.xxx.xx	TCP	54	59450 → 443 [ACK] Seq=1 Ack=1 Win=131072 Len=0
662	8.436082	192.168.70.10	xxx.xx.xxx.xx	TLSv1.2	384	Client Hello
667	8.643338	xxx.xx.xxx.xx	192.168.70.10	TCP	60	443 → 59450 [ACK] Seq=1 Ack=331 Win=64128 Len=0
668	8.645912	xxx.xx.xxx.xx	192.168.70.10	TLSv1.2	1466	Server Hello
669	8.646088	xxx.xx.xxx.xx	192.168.70.10	TLSv1.2	2878	Certificate [TCP segment of a reassembled PDU]
670	8.646088	xxx.xx.xxx.xx	192.168.70.10	TLSv1.2	160	Server Key Exchange, Server Hello Done
671	8.646177	192.168.70.10	xxx.xx.xxx.xx	TCP	54	59450 → 443 [ACK] Seq=331 Ack=4237 Win=131072 Len=0
672	8.646344	192.168.70.10	xxx.xx.xxx.xx	TCP	54	59450 → 443 [ACK] Seq=331 Ack=4343 Win=131072 Len=0
673	8.655383	192.168.70.10	xxx.xx.xxx.xx	TCP	54	59450 → 443 [FIN, ACK] Seq=331 Ack=4343 Win=131072 Len=0
674	8.873433	xxx.xx.xxx.xx	192.168.70.10	TCP	60	443 → 59450 [FIN, ACK] Seq=4343 Ack=332 Win=64128 Len=0
675	8.873611	192.168.70.10	xxx.xx.xxx.xx	TCP	54	59450 → 443 [ACK] Seq=332 Ack=4344 Win=131072 Len=0

I am not overly familiar with TLS, but I can't see any errors here. It looks like the client abruptly resets after receiving the Server Hello Done message. The service will continue to attempt this connection multiple times before failing. We have run this same workload in a standalone Docker container with the same OS image and it connects successfully.

We can't be sure that Envoy is causing the issue here, but it's the main difference in configuration between the two systems. Are there any reasons why this would happen? Are we potentially missing a configuration, etc.?

**Comment by ryan-motive** [comment]
In this case Envoy is the TLS client. We've found some documentation for the service/SDK we're using that it may be related to certificate checks failing. Not clear to me why Envoy would make a difference, though.

We've tried quite a few configurations, including trying to disable egress altogether. In any case the presence of Envoy seems to be the only common factor. Personally it really seems like Envoy is a red herring, but I wondered if there were any known issues with Envoy and TLS that might impact this.
